### PR TITLE
laurent: driver for KernelChip Laurent family of relays

### PR DESCRIPTION
**Pull request by lumag**
Add support for the KernelChip Laurent-2 / Laurent-5 / Laurent-112 / Laurent-128 network-controlled relay arrays.

### CODE REVIEW

File: schema.yaml [hunk 1]
```diff
@@ -142,6 +142,23 @@ properties:
...
               $ref: "#/$defs/local_gpio"
+
+        laurent:
+          description: KernelChip Laurent relays
+          type: object
+          unevaluatedItems: false
+          properties:
+            server:
+              type: string
+            relay:
```
Comment (superna9999) [inline schema.yaml:153]: Is this the relay port number ? Perhaps use something like "port" instead

Comment (lumag) [reply schema.yaml:153]: I don't want for it to be confused with the TCP port on which to talk to the server.

File: drivers/laurent.c [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+
+	laurent_resolve(laurent);
+
+	return laurent;
+}
+
+static int laurent_power(struct device *dev, bool on)
+{
+	struct laurent *laurent = dev->cdb;
+	char buf[BUFSIZ];
```
Comment (konradybcio) [inline drivers/laurent.c:130]: I suppose it's safe to assume nobody has a 470+ bytes long password

**Review by konradybcio — APPROVED** [review]
lgtm

File: drivers/laurent.c (hunk 2 above)
Comment (lumag) [reply drivers/laurent.c:130]: Don't give me wrong ideas!

**Pull request merged**